libs/libarchfpga/src/physical_types_util.h

```diff
@@ -0,0 +1,132 @@
+#ifndef PHYSICAL_TYPES_UTIL_H
+#define PHYSICAL_TYPES_UTIL_H
+
+#include "physical_types.h"
+
+/********************************************************************
+ *                                                                  * 
+ *  Physical types utility functions                                *
+ *                                                                  *
+ *  This source file contains several utilities that enable the     *
+ *  interaction with the architecture's physical types.             *
+ *  Mainly, the two classes of objects accessed by the utility      *
+ *  functions in this file are the following:                       *
+ *    - physical_tile_type                                          *
```

Comment:
Probably good to explain what a physical_tile_type is (place on the grid I think) vs. a logical_block_type (netlist item I think).  Are logical_block_types only in the clb_netlist or can they be in either the atom_netlist or clb_netlist?

Reply:
I think we need a big comment that defines logical_pin_index, physical_in_index, sub_tile, capacity etc. Can refer to a block comment in another file if we already have all that or otherwise define those terms.  Those terms are used in comments below, but without clear definitions and an example of how they're used its hard to understand the comments and notation in the routines.

Reply:
Done